### PR TITLE
Model deployment steps in RAML.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
This replaces the last custom deployment format and uses
the Jackson serialization.

### CODE REVIEW

**Comment by jeschkies** [comment]
Ping @timcharper and @ANeumann82

File: docs/docs/rest-api/public/api/v2/types/deploymentPlan.raml [hunk 1]
```diff
@@ -15,7 +15,7 @@ types:
         type: string
         description: The UUID of the deployment plan
       steps:
-        type: deploymentSteps.DeploymentSteps
+        type: deploymentSteps.DeploymentStep[]
```
Comment (ANeumann82) [inline docs/docs/rest-api/public/api/v2/types/deploymentPlan.raml]: This is a breaking API change, isn't it? We now have:
```
"steps": [
   { ...step... },
   { ...step...},
]
```
intead of:
```
"steps": {
  "steps": [
    { ...step...},
    { ...step...}
  ]
}
```

Or do I read that wrong?

Comment (jeschkies) [reply docs/docs/rest-api/public/api/v2/types/deploymentPlan.raml]: No. I had to look this up. `deploymentSteps.DeploymentStep` has no `steps` field but `deploymentSteps.DeploymentStepS` has.

**Comment by jeschkies** [comment]
Ping @timcharper and @ANeumann82

File: docs/docs/rest-api/public/api/v2/types/deploymentPlan.raml (hunk 1 above)
Comment (timcharper) [reply docs/docs/rest-api/public/api/v2/types/deploymentPlan.raml]: I see the same thing as @ANeumann82 ; the RAML was actually wrong but we weren't using the RAML to serialize the deployment plan (https://github.com/mesosphere/marathon/pull/7081/files#diff-df71fde3a131a175b64c40a758c0423dL203)

**Pull request merged**